sync/task_pool.go

```diff
@@ -158,14 +159,54 @@ func (p *TaskPool) run(id int, q chan task) error {
 	}
 }
 
-// add task
+// AddTask wait idle worker add task
 func (p *TaskPool) AddTask(t task) {
+	idx := atomic.AddUint32(&p.idx, 1)
+	id := idx % uint32(p.tQNumber)
+
+	select {
+	case <-p.done:
+		return
+	case p.qArray[id] <- t:
+	}
+}
+
+// AddTaskAlways add task to queues or do it immediately
+func (p *TaskPool) AddTaskAlways(t task) {
 	id := atomic.AddUint32(&p.idx, 1) % uint32(p.tQNumber)
 
 	select {
 	case <-p.done:
 		return
 	case p.qArray[id] <- t:
+		return
+	default:
+		go t()
+	}
+}
+
+// AddTaskBalance add task to idea queue
+// do it immediately when no idea queue
+func (p *TaskPool) AddTaskBalance(t task) {
+	length := len(p.qArray)
+
+	// try len/2 times to lookup idea queue
+	for i := 0; i*2 < length; i++ {
+		i := rand.Intn(length)
```

Comment:
use variable `i` again, change it to another

Reply:
fixed